scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,624 @@
+#!/usr/bin/env python3
+
+import argparse
+import fnmatch
+import json
+import os
+import subprocess
+
+from typing import Any, List, Optional, Union
+
+import colorama # Enables the use of SGR & CUP terminal VT sequences on Windows.
+from deepdiff import DeepDiff
+
+# {{{ JsonRpcProcess
+class BadHeader(Exception):
+    def __init__(self, msg: str):
+        super().__init__("Bad header: " + msg)
+
+class MyEncoder(json.JSONEncoder):
+    """
+    Encodes an object in JSON
+    """
+    def default(self, o):
+        return o.__dict__
+
+class JsonRpcProcess:
+    exe_path: str
+    exe_args: List[str]
+    process: subprocess.Popen
+    trace_io: bool
+
+    def __init__(self, exe_path: str, exe_args: List[str], trace_io: bool = True):
+        self.exe_path = exe_path
+        self.exe_args = exe_args
+        self.trace_io = trace_io
+
+    def __enter__(self):
+        self.process = subprocess.Popen(
+            [self.exe_path, *self.exe_args],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback) -> None:
+        self.process.kill()
+        self.process.wait(timeout=2.0)
+
+    def trace(self, topic: str, message: str) -> None:
+        if self.trace_io:
+            print(f"{SGR_TRACE}{topic}:{SGR_RESET} {message}")
+
+    def receive_message(self) -> Union[None, dict]:
+        # Note, we should make use of timeout to avoid infinite blocking if nothing is received.
+        CONTENT_LENGTH_HEADER = "Content-Length: "
+        CONTENT_TYPE_HEADER = "Content-Type: "
+        if self.process.stdout == None:
+            return None
+        message_size = None
+        while True:
+            # read header
+            line = self.process.stdout.readline()
+            if line == '':
+                # server quit
+                return None
+            line = line.decode("utf-8")
+            if not line.endswith("\r\n"):
+                raise BadHeader("missing newline")
+            # remove the "\r\n"
+            line = line[:-2]
+            if line == "":
```

Comment:
```suggestion
            if line == '':
```
(or just `if not line` - at least it should be the same as in line 64)

Reply:
The `if x:` syntax is dangerous. A very common mistake is to forget that you're checking an int variable that could contain `0` (which evaluates to false). I even caught one of these in this very PR.